The build failed on i386 fails maybe because the macro `JSC_HOST_CALL` is not set correctly. The error output is:
```
../Source/WebCore/bindings/js/JSDOMConstructor.h:195:35: error: invalid conversion from ‘JSC::EncodedJSValue (*)(JSC::ExecState*)’ {aka ‘long long int (*)(JSC::ExecState*)’} to ‘JSC::NativeFunction’ {aka ‘long long int (__attribute__((fastcall)) *)(JSC::ExecState*)’} [-fpermissive]
     constructData.native.function = construct;
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
```

`construct` is declared as 
```
static JSC::EncodedJSValue JSC_HOST_CALL construct(JSC::ExecState*);
```
`JSC_HOST_CALL` seems not to be correct. It's defined in `webkit/Source/WTF/wtf/Platform.h`:
```
#if CPU(X86) && COMPILER(MSVC)
#define JSC_HOST_CALL __fastcall
#elif CPU(X86) && COMPILER(GCC_OR_CLANG)
#define JSC_HOST_CALL __attribute__ ((fastcall))
#else
#define JSC_HOST_CALL
#endif
```
Maybe some changes in the Freedesktop SDK cause this error?

--- a/io.qt.qtwebkit.BaseApp.json
+++ b/io.qt.qtwebkit.BaseApp.json
@@ -41,6 +41,10 @@
         },
         {
           "type": "patch",
+          "path": "patches/fix_i386_detection.patch"
+        },
+        {
+          "type": "patch",
           "path": "patches/fix_selectedHtml_segfault.diff"
         },
         {

--- a/patches/fix_i386_detection.patch
+++ b/patches/fix_i386_detection.patch
@@ -1,0 +1,12 @@
+Description: fix x32 detection
+
+--- a/Source/WTF/wtf/Platform.h
++++ b/Source/WTF/wtf/Platform.h
+@@ -158,6 +158,7 @@
+ /* CPU(X86) - i386 / x86 32-bit */
+ #if   defined(__i386__) \
+     || defined(i386)     \
++    || defined(__i386)   \
+     || defined(_M_IX86)  \
+     || defined(_X86_)    \
+     || defined(__THW_INTEL)

